packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -0,0 +1,1112 @@
+    this._requirePublicClient()
+
+    const [createLogs, owner, paused] = await Promise.all([
+      this._publicClient?.getLogs({
+        event: splitCreatedEvent,
+        address: [
+          getSplitV2FactoryAddress(this._chainId, SplitV2Type.Pull),
+          getSplitV2FactoryAddress(this._chainId, SplitV2Type.Push),
+        ],
+        args: {
+          split: splitAddress,
+        },
+        strict: true,
+        fromBlock: getSplitV2FactoriesStartBlock(this._chainId),
+      }),
+      this._owner(splitAddress),
+      this._paused(splitAddress),
+    ])
+
+    if (!createLogs) throw new Error('Split not found')
+
+    const updateLogs = await this._publicClient?.getLogs({
+      address: splitAddress,
+      event: splitUpdatedEvent,
+      strict: true,
+      fromBlock: createLogs[0].blockNumber,
+    })
+
+    const recipients = createLogs[0].args.splitParams.recipients.map(
+      (recipient, i) => {
+        return {
+          recipient: {
+            address: recipient,
+          },
+          percentAllocation: fromBigIntToPercent(
+            createLogs[0].args.splitParams.allocations[i],
+            createLogs[0].args.splitParams.totalAllocation,
+          ),
+        }
+      },
+    )
+
+    const split: Split = {
+      address: splitAddress,
+      recipients,
+      distributorFeePercent: fromBigIntToPercent(
+        createLogs[0].args.splitParams.distributionIncentive,
+      ),
+      distributeDirection: getSplitType(this._chainId, createLogs[0].address),
+      type: 'SplitV2',
+      controller: {
+        address: owner,
+      },
+      distributionsPaused: paused,
+      newPotentialController: {
+        address: zeroAddress,
+      },
+      createdBlock: Number(createLogs[0].blockNumber),
+    }
+
+    if (!updateLogs || updateLogs.length == 0) return { split }
+
+    updateLogs.sort((a, b) => {
+      if (a.blockNumber !== b.blockNumber)
+        return a.blockNumber > b.blockNumber ? -1 : 1
+      else return a.logIndex > b.logIndex ? -1 : 1
+    })
+
+    const updatedRecipients = updateLogs[0].args._split.recipients.map(
+      (recipient, i) => {
+        return {
+          recipient: {
+            address: recipient,
+          },
+          percentAllocation: fromBigIntToPercent(
+            updateLogs[0].args._split.allocations[i],
+            updateLogs[0].args._split.totalAllocation,
+          ),
+        }
+      },
+    )
+
+    split.recipients = updatedRecipients
+    split.distributorFeePercent = fromBigIntToPercent(
+      updateLogs[0].args._split.distributionIncentive,
+    )
+
+    return { split }
+  }
+
+  protected _getSplitV2Contract(
+    splitAddress: Address,
+  ): GetContractReturnType<SplitV2ABI, PublicClient<Transport, Chain>> {
+    validateAddress(splitAddress)
+
+    return getContract({
+      address: splitAddress,
+      abi: splitV2ABI,
+      publicClient: this._publicClient,
+      walletClient: this._walletClient,
```

Comment:
nit: need to have a `client` option here to support viem v2. here's what i did elsewhere:

```return getContract({
      address: getAddress(liquidSplit),
      abi: ls1155CloneAbi,
      // @ts-expect-error v1/v2 viem support
      client: this._publicClient,
      publicClient: this._publicClient,
    })

```

Reply:
could modify that to still support both public and wallet client. but i think `client` is needed for viem v2. but it can take both public or wallet client